### DISCUSSION TITLE
Meeting 5 - Feb 8th

### DISCUSSION THREAD

**Discussion by robertlong** [omigroup/omi-scripting-group#20] [General]
# Agenda

- Updates on the experiments
- API design
- High level WASM UGC concepts

**Comment by Trensharo** [comment]
Last Meeting- Discussed experimental projects to test simple concepts.

**cdata**
- Working on a project embedding WebAssembly
- 3D Renderer that runs with parody in web browser and as a native binary
- Hot load in browser and native binary
- Wants to try to homogenize the spaces

**Robert**
- Nate has been working on DOD integration library - https://github.com/Web-ECS/dod-three
- Robert wants to build on top of this project
- Wants to get started on this within a WASM concept

**Reply by Trensharo** [reply]
What to do with these projects?

- Come up with useful applications of the base proofs of concept.
- Have others try to apply the code in a different context to spur feedback and ideation.

- Presentations can include barebones demos of how to actually create these WASM scripts

How to potentially apply this with engines such as Unity?
- For C# - Blazor is a web app feature - https://dotnet.microsoft.com/en-us/apps/aspnet/web-apps/blazor
-  -Robert provided this info
-  -This may be helpful for those with Unity and C# experience
- Take a look at VRChat's UdonSharp compiler for an example of application of a similar concept - https://github.com/vrchat-community/UdonSharp
-  -Robert provided this info
- There is an experimental project by dotnet that works with C# as a WASM compiler - https://github.com/dotnet/runtimelab/tree/feature/NativeAOT-LLVM
- -Adam provided this info
- https://medium.com/@scottwaye/doom-fire-in-the-browser-using-c-12128c3f4c94

**Reply by Trensharo** [reply]
What are the WASM engines?
- https://github.com/appcypher/awesome-wasm-runtimes
- Plenty of options to choose from
- Wasmtime and Wasmer are likely the most supported.

**Reply by Trensharo** [reply]
**Recruitment**
- We could do a social media push
- Maybe get a video created
- Put up embedded demos of work - Mozilla hubs, webiverse

**Potential Collaborators to invite**
- Feather Devs - Minecraft Server with WASM integration - https://feathermc.org/
- Neos Devs - https://neos.com/
- Mozilla Devs?

**Reply by Trensharo** [reply]
Jin will share a discord event for Friday/Saturday about joining with a Neos meetup
